lib/jsdom/living/xhr-utils.js

```diff
@@ -113,8 +115,9 @@ function requestErrorSteps(xhr, event, exception) {
     properties.uploadComplete = true;
 
     if (properties.uploadListener) {
-      fireAnEvent(event, xhr.upload, ProgressEvent, { loaded: 0, total: 0, lengthComputable: false });
-      fireAnEvent("loadend", xhr.upload, ProgressEvent, { loaded: 0, total: 0, lengthComputable: false });
+      const uploadWrapper = idlUtils.tryImplForWrapper(xhr.upload);
```

Comment:
The name seems wrong here; should it be uploadImpl? In general not sure why this changed; was it a bug you found?

Reply:
Currently, `XMLHttpRequest` doesn't uses wedidl2js. All the properties including [`upload`](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xmlhttprequest.js#L102) are wrappers and not impls. With this PR the `fireAnEvent` function expects an impl and not a wrapper, so I need to unwrap before the invocation.

I initially undertook the task of converting the `XMLHttpRequest` interface to leverage webidl2js, however it was too much change to fit into a single PR. Once this PR is merged, I need to open couple of issues to remove some of the hacks now that all the interfaces have access to the global object.

Reply:
That makes sense, I'm just confused why the variable name is uploadWrapper if you're calling (try)ImplForWrapper, which returns an impl.